llvm/include/llvm/ProfileData/MemProfData.inc

```diff
@@ -90,111 +93,139 @@ PACKED(struct SegmentEntry {
-  MinLifetime = TotalLifetime;
-  MaxLifetime = TotalLifetime;
-  // Access density is accesses per byte. Multiply by 100 to include the
-  // fractional part.
-  TotalAccessDensity = AccessCount * 100 / Size;
-  MinAccessDensity = TotalAccessDensity;
-  MaxAccessDensity = TotalAccessDensity;
-  // Lifetime access density is the access density per second of lifetime.
-  // Multiply by 1000 to convert denominator lifetime to seconds (using a
-  // minimum lifetime of 1ms to avoid divide by 0. Do the multiplication first
-  // to reduce truncations to 0.
-  TotalLifetimeAccessDensity =
-      TotalAccessDensity * 1000 / (TotalLifetime ? TotalLifetime : 1);
-  MinLifetimeAccessDensity = TotalLifetimeAccessDensity;
-  MaxLifetimeAccessDensity = TotalLifetimeAccessDensity;
-  AllocCpuId = AllocCpu;
-  DeallocCpuId = DeallocCpu;
-  NumMigratedCpu = AllocCpuId != DeallocCpuId;
-}
-
-void Merge(const MemInfoBlock &newMIB) {
-  AllocCount += newMIB.AllocCount;
-
-  TotalAccessCount += newMIB.TotalAccessCount;
-  MinAccessCount = newMIB.MinAccessCount < MinAccessCount ? newMIB.MinAccessCount : MinAccessCount;
-  MaxAccessCount = newMIB.MaxAccessCount > MaxAccessCount ? newMIB.MaxAccessCount : MaxAccessCount;
-
-  TotalSize += newMIB.TotalSize;
-  MinSize = newMIB.MinSize < MinSize ? newMIB.MinSize : MinSize;
-  MaxSize = newMIB.MaxSize > MaxSize ? newMIB.MaxSize : MaxSize;
-
-  TotalLifetime += newMIB.TotalLifetime;
-  MinLifetime = newMIB.MinLifetime < MinLifetime ? newMIB.MinLifetime : MinLifetime;
-  MaxLifetime = newMIB.MaxLifetime > MaxLifetime ? newMIB.MaxLifetime : MaxLifetime;
-
-  TotalAccessDensity += newMIB.TotalAccessDensity;
-  MinAccessDensity = newMIB.MinAccessDensity < MinAccessDensity
-                         ? newMIB.MinAccessDensity
-                         : MinAccessDensity;
-  MaxAccessDensity = newMIB.MaxAccessDensity > MaxAccessDensity
-                         ? newMIB.MaxAccessDensity
-                         : MaxAccessDensity;
-
-  TotalLifetimeAccessDensity += newMIB.TotalLifetimeAccessDensity;
-  MinLifetimeAccessDensity =
-      newMIB.MinLifetimeAccessDensity < MinLifetimeAccessDensity
-          ? newMIB.MinLifetimeAccessDensity
-          : MinLifetimeAccessDensity;
-  MaxLifetimeAccessDensity =
-      newMIB.MaxLifetimeAccessDensity > MaxLifetimeAccessDensity
-          ? newMIB.MaxLifetimeAccessDensity
-          : MaxLifetimeAccessDensity;
-
-  // We know newMIB was deallocated later, so just need to check if it was
-  // allocated before last one deallocated.
-  NumLifetimeOverlaps += newMIB.AllocTimestamp < DeallocTimestamp;
-  AllocTimestamp = newMIB.AllocTimestamp;
-  DeallocTimestamp = newMIB.DeallocTimestamp;
-
-  NumSameAllocCpu += AllocCpuId == newMIB.AllocCpuId;
-  NumSameDeallocCpu += DeallocCpuId == newMIB.DeallocCpuId;
-  AllocCpuId = newMIB.AllocCpuId;
-  DeallocCpuId = newMIB.DeallocCpuId;
-}
+  }
+
+  MemInfoBlock(uint32_t Size, uint64_t AccessCount, uint32_t AllocTs,
```

Comment:
There are a lot of formatting only changes in the patch (across a number of files). Can you revert those?

Reply:
This is all done by clang-format. Will it not fail formatting check if I change this?

Reply:
I don't think that would be a hard failure. Not sure why it wasn't clang formatted before, but better to stick with the format of the surrounding code. Generally it won't reformat the whole file, though, only your changed lines. How did you invoke it?

Reply:
It automatically runs in IDE every time I save a document, and then I invoked it manually before creating pull request with `git clang-format HEAD~1`, but that didn't change anything.

Reply:
Ok, let me remove the formatting.

Reply:
There's probably a way to disable that if it is reformatting the whole file when you save. For command line reformatting I do `git clang-format --style LLVM -f` which only reformats my changed lines

Reply:
Thanks

Reply:
According to `git clang-format -h`, `git clang-format HEAD~1` should only reformat things touched in the most recent commit, so perhaps it is the IDE that is reformatting the whole file?